aten/src/ATen/core/jit_type.h

```diff
@@ -710,24 +687,25 @@ struct CAFFE2_API FloatType : public Type {
   std::string str() const override {
     return "float";
   }
+  std::string python_str() const override {
```

Comment:
these are unnecessary in the cases where str and python_str are the same, right?

Reply:
I think this is because we recently added a `python_str()` override in `NumberType`. Now FloatType and IntType are inheriting `NumberType` rather than `Type`, so `FloatType` will call `NumberType`'s python_str() instead of the `Type`'s virtual `python_str()` if its own python_str() is not present.

Reply:
So I think BoolType and NoneType `python_str()` are not needed and I can delete them, but `IntType` and `FloatType` will still need that if we want the `NumberType` to be their base type.

Reply:
makes sense